src/ray/common/scheduling/scheduling_ids.h

```diff
@@ -0,0 +1,50 @@
+#ifndef RAY_COMMON_SCHEDULING_SCHEDULING_IDS_H
+#define RAY_COMMON_SCHEDULING_SCHEDULING_IDS_H
+
+#include <iostream>
+#include <map>
+#include <unordered_map>
+#include <string>
+#include <iterator>
+#include <stdlib.h>
+#include <chrono>
+#include <vector>
+using namespace std;
+
+/// Class to map string IDs to unique integer IDs and back.
+class ScheduleIds {
+  unordered_map<string, int64_t> string_to_int;
+  unordered_map<int64_t, string> int_to_string;
+  hash<string> hasher;
+
+public:
+  ScheduleIds() {};
+  ~ScheduleIds() {};
+
+  /// Get integer ID associated with an existing string ID.
+  ///
+  /// \param String ID.
+  /// \return The integer ID associated with the given string ID.
+  int64_t getIdByInt(string sid);
```

Comment:
I'd suggest these signatures for the methods:

`int64_t get(const std::string& resource_label)`
`int64_t insert(const std::string& resource_label)`
`void remove(const std::string& resource_label)`
`void remove(int64_t resource_id)`

Reply:
done